movie.js

```diff
@@ -0,0 +1,131 @@
+import { movies } from "./storemovies.js";
+
+//const movieComments = {};
+
+function createMovieCard(movie) {
+  const card = document.createElement("div");
+  card.classList.add("movie-card");
+
+  card.innerHTML = `
+    <img src="${movie.poster_url}" alt="${movie.title} Poster" class="movie-poster">
+    <div class="star-rating" data-movie-id="${movie.id}">
+      <div>
+        <i class="fa-solid fa-circle-info info-icon" style="color: #74C0FC;"></i>
+      </div>
+      <div>
+        <button class="star">&#9734;</button>
+        <button class="star">&#9734;</button>
+        <button class="star">&#9734;</button>
+        <button class="star">&#9734;</button>
+        <button class="star">&#9734;</button>
+      </div>
+    </div>
+  `;
+
+  const infoIcon = card.querySelector(".info-icon");
+  infoIcon.addEventListener("click", function (event) {
+    event.stopPropagation(); // Prevent the card click event from firing
+    showMovieInfo(movie);
+  });
+
+  return card;
+}
+
+function showMovieInfo(movie) {
+  const modal = document.getElementById("movieModal");
+  const modalContent = document.getElementById("modal-movie-info");
+
+  modalContent.innerHTML = `
+    <h2 class="movie-title">${movie.title}</h2>
+    <p class="movie-description">${movie.description}</p>
+    <p class="movie-details"><strong>Year:</strong> ${movie.movie_year}</p>
+    <p class="movie-price">$${movie.price}</p>
+    <form id="comment-form-${movie.id}">
+    <input type="text" placeholder="comment here ..." class="input-comment" id="input-comment-${movie.id}"/>
+    <button type="submit" class="btn" id="submit-comment-${movie.id}">Submit</button>
+    </form>
+    <h4 class="comment-heading">Comments:</h4>
+    <div class="comment-list" id="comment-list-${movie.id}"></div>
+  `;
+
+  modal.style.display = "block";
+
+  const closeModal = document.querySelector(".close");
+  closeModal.onclick = function () {
+    modal.style.display = "none";
+  };
+
+  const commentForm = document.getElementById(`comment-form-${movie.id}`);
+  commentForm.addEventListener("submit", function (event) {
+    event.preventDefault();
+    addComment(movie.id, movie.comments);
+  });
+
+  displayComments(movie.id, movie.comments);
+
+  window.onclick = function (event) {
+    if (event.target == modal) {
+      modal.style.display = "none";
+    }
+  };
+}
+
+function addComment(movieId, movieComment) {
+  const commentInput = document.getElementById(`input-comment-${movieId}`);
```

Comment:
IMO this function should be agnostic of the input. You should retrieve the text value in the `submit` event handler and pass it to this function. 

In pseudo code:
```js
function addComment(movieId, commentText) { 
  // 1. Find the movie that matches the movieId
  // 2. Check that the movie exists
  // 3. Add the comment to the movie's comments list
}
```